src/pages/blog-cn.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react'
+import { Router, useLocation } from '@reach/router'
+import { navigate } from 'gatsby'
+
+const BlogCN = () => {
+  const location = useLocation()
+  const blogTag = location.hash.slice(1)
+  const blogPathName = location.pathname.split('/')[2]
+
+  const RedirectBlogCNPath = () => {
+    // redirect /blog-cn/#tag
+    if (blogTag) {
+      navigate(`/zh/blog/tag/${blogTag}`)
+    } else if (blogPathName) {
+      // redirect /blog-cn/BLOG_PATH
+      navigate(`/zh/blog/${blogPathName}`)
+    } 
+    return null
+  }
+
+  return (
+    <Router basepath="/blog-cn">
```

Comment:
Could you explain why use `Router` here?

I think maybe this can be replaced with by `useEffect` and `return null`, for example:

```js
useEffect(() => {
  redirect()
}, [])

return null
```

Reply:
Both methods are acceptable. The `Router` and `RedirectBlogCNPath` component just use to match the pathname and redirect to corresponding page.

Reply:
Know about it.